ops/model.py

```diff
@@ -334,9 +450,24 @@ def network(self):
 
 
 class Network:
-    """Network space details."""
+    """Network space details.
+
+    Attributes:
+        interfaces: A list of :class:`NetworkInterface` details. This includes the
+            information about how your application should be configured (eg, what
+            IP addresses should you bind to.)
+            Note that multiple addresses for a single interface are represented as multiple
+            interfaces. (eg, ``[NetworKInfo('ens1', '10.1.1.1/32'),
+            NetworkInfo('ens1', '10.1.2.1/32'])``)
+        ingress_addresses: A list of :class:`ipaddress.ip_address` objects representing the IP
+            addresses that other units should use to get in touch with you.
+        egress_subnets: A list of :class:`ipaddress.ip_network` representing the subnets that
+            other units will see you connecting from. Due to things like NAT it isn't always
+            possible to narrow it down to a single address, but when it is clear, the CIDRs
+            will be constrained to a single address. (eg, 10.0.0.1/32)
+    """
 
-    def __init__(self, network_info):
+    def __init__(self, network_info: dict):
```

Comment:
without a `Parameters` (or its aliases `Args`, `Arguments`), either here or in the class's docstring, the documentation will be lacking:
![image](https://user-images.githubusercontent.com/1543430/82198736-a2438e80-98f4-11ea-8668-00ae4997850a.png)

in some classes it isn't a big deal because the parameters and the attributes are the same so it's reasonable to assume they're the same thing, but frequently not.

The default sphinx class makes it a bigger issue than the RTD one, as the spacing is out of whack, but that only means that it's easier to spot with the RTD theme :-)

Reply:
So I was not documenting __init__ for classes that we didn't want users creating directly.

That said, if it looks bad in the docs, we should make it better.
I also added the "not instantiate directly" to this class.

Reply:
Hmm... maybe there's a way for it not to enumerate the args to init for the cases we don't want people to instantiate?